game/world/objects/npc.cpp

```diff
@@ -2830,6 +2830,24 @@ const Npc::Routine& Npc::currentRoutine() const {
       return i;
     }
 
+  // take previous routine if none is set
+  const auto     day   = gtime(24,0).toInt();
```

Comment:
This code probably duplicated one from `resetPositionToTA`
```
  if(at==nullptr) {
    const auto time  = owner.time().timeInDay();
    const auto day   = gtime(24,0).toInt();
    int64_t    delta = std::numeric_limits<int64_t>::max();

    // closest time-point
    for(auto& i:routines) {
   ...
```
Need to generalize, and add handling of null-waypooint here